src/edu/cmu/tartan/Game.java

```diff
@@ -248,7 +268,34 @@ public String getGameDescription(){
      */
     public boolean validate() {
         // TODO: This method is way too simple. A more thorough validation must be done!
-        return (gameName != null && gameDescription != null);
+		if(this.goals instanceof GameCollectGoal){
+			List<String> itemsListGoal = ((GameCollectGoal) this.goals).getItemsList();
+			int numItem = 0;
+
+			for (Room i : this.roomArrayList){
+				numItem += i.items.size();
+			}
+
+			return itemsListGoal.size() > numItem;
+		}
+		else if(this.goals instanceof GameExploreGoal){
+			List<String> itineraryListGoal = ((GameExploreGoal) this.goals).getItinerary();
+
+			return itineraryListGoal.size() > this.roomArrayList.size();
+		}
+		else if(this.goals instanceof GamePointsGoal){
+			int score = 0;
+			for (Room i : this.roomArrayList){
+				for (Item j : i.items){
+					score += j.value();
+				}
+			}
+
+			return ((GamePointsGoal) this.goals).getWinningScore() > score;
+		}
+		else{
+			return true;
+		}
```

Comment:
1) If there are more than 2 targets, it can not be detected under the condition.
2) No treatment for no goal
3) In the case of point games, you may be able to get points from items, but you have to make sure that you do not get points in any other cases.
4) In case of item game, check if there is no corresponding item. If you check only the number, the game will not end if an item that is missing is targeted.